addons/account/models/partner.py

```diff
@@ -168,11 +168,10 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
         # This can be easily overridden to apply more complex fiscal rules
         PartnerObj = self.env['res.partner']
         partner = PartnerObj.browse(partner_id)
+        delivery = PartnerObj.browse(delivery_id)
 
-        # if no delivery use invoicing
-        if delivery_id:
-            delivery = PartnerObj.browse(delivery_id)
-        else:
+        # If partner and delivery have the same vat prefix, use invoicing
+        if not(delivery.vat and partner.vat and delivery.vat[:2] == partner.vat[:2]):
```

Comment:
Maybe we were too quick?
```suggestion
        if not delivery.vat or (partner.vat and delivery.vat[:2] == partner.vat[:2]):
```

Reply:
Indeed :sweat_smile:

Reply:
Indeed :sweat_smile: